### PR TITLE
Makefile check_tools enhancement to cover more external dependencies

### PR DESCRIPTION
**Pull request by Greg-Szabo**
This change adds features to the `check_tools` target of the `Makefile`.
- Checks if go is installed
- Checks the minimum required go version
- Checks if GOPATH is set
- Checks if PATH contains GOPATH/bin (still discussed in issue #966)

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -35,9 +36,17 @@ install:
 ### Tools & dependencies
 
 check_tools:
-	@# https://stackoverflow.com/a/25668869
-	@echo "Found tools: $(foreach tool,$(GOTOOLS_CHECK),\
-        $(if $(shell which $(tool)),$(tool),$(error "No $(tool) in PATH")))"
+	@echo "Checking go availability"
+	@test -x $(shell which go)
+	@echo "Checking go version $(GO_VERSION) >= $(GO_MIN_VERSION)"
+	@test $(shell echo $(GO_VERSION) | grep -o '^[[:digit:]]\+' ) -ge $(shell echo $(GO_MIN_VERSION) | grep -o '^[[:digit:]]\+' )
```
Comment (melekes) [inline Makefile]: Do we really need this? Can't we just say that required go version is 1.9 in README?

Comment (melekes) [reply Makefile]: I'd vote against this PR. It brings additional complexity and I am not sure we need it.

Comment (Greg-Szabo) [reply Makefile]: I'm trying to minimize the amount of documentation required for the build process. Documentation can get obsolete but automation is tested with every commit.

**Comment by zramsay** [comment]
yeah this would help a lot with jenkins - just ran into issues solved with this PR

**Comment by Greg-Szabo** [comment]
Test comment from Jenkins, please ignore.

**Comment by Greg-Szabo** [comment]
Coming up with a better method, I'm closing this.

File: Makefile (hunk 1 above)
Comment (Greg-Szabo) [reply Makefile]: A better version was created. It's pull requested at #1042

**Pull request closed without merging**